pkg/agent/reconcilers.go

```diff
@@ -174,8 +173,7 @@ func (a *Agent) updateWorkStatus(obj runtime.Object, isBeingDeleted bool) error
 			}
 
 			// only update status for KS-managed (by bindingpolicies here) objects
-			// TODO remove the check for ManagedByKSLabelKeyPrefix after we switch to new transport
-			if !util.HasPrefixInMap(manifestWork.Labels, ManagedByKSLabelKeyPrefix) && !util.HasPrefixInMap(manifestWork.Labels, TransportLabelPrefix) {
+			if !util.HasPrefixInMap(manifestWork.Labels, TransportKSLabelKeyPrefix) {
```

Comment:
in the new implementation the labels are added on the workload objects and not on manifest work.
why do we check manifestWork.Labels? what am I missing?

Reply:
Here we filter out the objects with ManifestWorks that was not created by KS (e.g. was not labeled with `transport.kubestellar.io/...`). This is not related to the singleton label on the object which is `managed-by.kubestellar.io/singletonstatus`